src/plugins/region_map/public/region_map_type.js

```diff
@@ -127,6 +186,10 @@ provided base maps, or add your own. Darker colors represent higher values.',
     setup: async (vis) => {
       const serviceSettings = await getServiceSettings();
       const tmsLayers = await serviceSettings.getTMSServices();
+
+      vis.http = http;
+      vis.params.http = http;
+      vis.notifications = notifications;
```

Comment:
Why are we adding these objects to `vis`. I cant see it being used from here within the code. The `vis` here is typed as `any` which is actually wrong and is why you dont see any typescript errors. The vis object is meant for properties of the visualization and its services. http, and notifications are OSD services that can be accessed using the `useOpenSearchDashboards` hook

Reply:
I was trying to pass these props from here to use in some components. But thanks for the pointer, I'm updating it now : )